doc/zconfig.rst

```diff
@@ -0,0 +1,1989 @@
+
+References to defined names from configuration values use the syntax
+described for the :mod:`ZConfig.substitution` module.
+Configuration values which include a ``$`` as part of the
+actual value will need to use ``$$`` to get a single
+``$`` in the result.
+
+The values of defined names are processed in the same way as
+configuration values, and may contain references to named
+definitions.
+
+For example, the value for ``key`` will evaluate to ``value``::
+
+
+  %define name value
+  key $name
+
+
+.. _writing-schema:
+
+Writing Configuration Schema
+============================
+
+:mod:`ZConfig` schema are written as XML documents.
+
+Data types are searched in a special namespace defined by the data
+type registry.  The default registry has slightly magical semantics:
+If the value can be matched to a standard data type when interpreted
+as a **basic-key**, the standard data type will be used.  If
+that fails, the value must be a **dotted-name** containing at
+least one dot, and a conversion function will be sought using the
+:meth:`~.search` method of the data type registry used to load the
+schema.
+
+.. _elements:
+
+Schema Elements
+---------------
+
+For each element, the content model is shown, followed by a
+description of how the element is used, and then a list of the
+available attributes.  For each attribute, the type of the value is
+given as either the name of a :mod:`ZConfig` datatype or an XML
+attribute value type.  Familiarity with XML's Document Type Definition
+language is helpful.
+
+The following elements are used to describe a schema:
+
+.. code-block:: xml
+
+ <schema>
+   description?, metadefault?, example?, import*, (sectiontype |
+   abstracttype)*, (section | key | multisection | multikey)*
+ </schema>
+
+Document element for a :mod:`ZConfig` schema.
+
+**extends** (**space-separated-url-references**)
+    A list of URLs of base schemas from which this section type will inherit
+    key, section, and section type declarations.  If omitted, this schema is
+    defined using only the keys, sections, and section types contained within
+    the ``schema`` element.
+
+**datatype** (**basic-key** or **dotted-name**)
+    The data type converter which will be applied to the value of this
+    section.  If the value is a **dotted-name** that begins
+    with a period, the value of ``prefix``  will be pre-pended,
+    if set.  If any base schemas are listed in the ``extends``
+    attribute, the default value for this attribute comes from the base
+    schemas.  If the base schemas all use the same ``datatype`` , then
+    that data type will be the default value for the extending schema.  If
+    there are no base schemas, the default value is **null** , which
+    means that the :mod:`ZConfig` section object will be used unconverted.
+    If the base schemas have different ``datatype``  definitions, you
+    must explicitly define the ``datatype``  in the extending schema.
+
+**handler** (**basic-key**)
```

Comment:
The only comment I could find was for SectionInfo (which extends the same base class that KeyInfo, etc does, which is where this attribute is defined) says "handler name called when value(s) must take effect, or None."

It doesn't appear to be used in this package, AFAICS. But the goal of this PR was to be a straightforward port to RST with additional cleanups later, so I didn't mess with it.